### PR TITLE
Update mentor profile UI

### LINKED ISSUE #171: Update mentor profile UI
**Is your feature request related to a problem? Please describe.**
The mentor profile page should highlight the number of total applications, available slots in a more user friendly way.

**Describe the solution you'd like**
Update mentor profile to closely resemble the following design:

image

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by AnsarMahir**
## Purpose

The purpose of this PR is to Update mentor profile UI #171

## Screenshots
image

##  Checklist
- [ ] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [ ] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Learning

closes #171

### CODE REVIEW

**Comment by anjula-sack** [comment]
@AnsarMahir The tests are failing can you check?

**Comment by AnsarMahir** [comment]
@anjula-sack The issue seems like package-lock and package-json was not in sync. did the necessary changes

**Comment by anjula-sack** [comment]
can you update this to version 18 in e2e.yml? @AnsarMahir

**Comment by AnsarMahir** [comment]
changed the node version to 18 from 16.  @anjula-sack

**Comment by anjula-sack** [comment]
Still failing @AnsarMahir. Can you remove the package-lock.json changes?

**Comment by anjula-sack** [comment]
@Madhawa97 Can you help @AnsarMahir with this actions failure?

**Comment by AnsarMahir** [comment]
@anjula-sack updated the package.json and package-lock file by replacing them with the files in the repo.

File: src/pages/MentorProfile/MentorProfile.component.tsx [hunk 1]
```diff
@@ -101,11 +101,51 @@ const MentorProfile: React.FC = () => {
...
+                            mentee.state === ApplicationStatus.APPROVED
+                        ).length
+                    ) === 0
+                      ? 'bg-gray-100 text-gray-800'
+                      : 'bg-green-100 text-green-800'
+                  }`}
+                >
+                  {Math.max(
+                    0,
+                    mentor.application.noOfMentees -
```
Comment (anjula-sack) [inline src/pages/MentorProfile/MentorProfile.component.tsx]: Create a variable for this. 

```suggestion
const noOfApprovedMentees = mentor.mentees.filter(
                        (mentee) => mentee.state === ApplicationStatus.APPROVED
                      ).length
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -102,4 +102,4 @@
       "plugin:storybook/recommended"
     ]
   }
-}
+}
```
Comment (anjula-sack) [inline package.json]: Add the end of line

```suggestion
}

```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/pages/MentorProfile/MentorProfile.component.tsx (hunk 1 above)
Comment (anjula-sack) [reply src/pages/MentorProfile/MentorProfile.component.tsx]:
```
// Calculate the number of approved mentees once
const approvedMenteesCount = mentor.mentees.filter(
  (mentee) => mentee.state === ApplicationStatus.APPROVED
).length;

// Calculate available slots
const availableSlots = Math.max(
  0,
  mentor.application.noOfMentees - approvedMenteesCount
);

// Render the result
<>
  {availableSlots}{' '}
  {availableSlots <= 1 ? 'Slot' : 'Slots'}{' '}
  Available
</>
```

**Comment by AnsarMahir** [comment]
@anjula-sack Do not merge

File: src/pages/MentorProfile/MentorProfile.component.tsx [hunk 3]
```diff
@@ -39,6 +39,18 @@ const MentorProfile: React.FC = () => {
...
   const { revokeApplication } = useMentee();
 
+  // Calculate the number of approved mentees once
+  const approvedMenteesCount = mentor?.mentees
+    ? mentor.mentees.filter(
+        (mentee) => mentee.state === ApplicationStatus.APPROVED
+      ).length
+    : 0;
+
+  // Calculate available slots
```
Comment (anjula-sack) [inline src/pages/MentorProfile/MentorProfile.component.tsx]:
```suggestion
```

File: src/pages/MentorProfile/MentorProfile.component.tsx [hunk 4]
```diff
@@ -39,6 +39,18 @@ const MentorProfile: React.FC = () => {
   const { data: mentor, isLoading } = useMentor(mentorId as string);
   const { revokeApplication } = useMentee();
 
+  // Calculate the number of approved mentees once
```
Comment (anjula-sack) [inline src/pages/MentorProfile/MentorProfile.component.tsx]:
```suggestion
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Comment by AnsarMahir** [comment]
@anjula-sack did the changes

File: src/pages/MentorProfile/MentorProfile.component.tsx [hunk 5]
```diff
@@ -56,6 +51,18 @@ const MentorProfile: React.FC = () => {
       .catch(() => {});
   };
 
+    // Calculate the number of approved mentees once
```
Comment (anjula-sack) [inline src/pages/MentorProfile/MentorProfile.component.tsx]:
```suggestion
```

File: src/pages/MentorProfile/MentorProfile.component.tsx [hunk 6]
```diff
@@ -56,6 +51,18 @@ const MentorProfile: React.FC = () => {
...
+    const approvedMenteesCount = mentor?.mentees
+    ? mentor.mentees.filter(
+        (mentee) => mentee.state === ApplicationStatus.APPROVED
+      ).length
+    : 0;
+
+  // Calculate available slots
```
Comment (anjula-sack) [inline src/pages/MentorProfile/MentorProfile.component.tsx]:
```suggestion
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/pages/MentorProfile/MentorProfile.component.tsx [hunk 7]
```diff
@@ -229,15 +241,6 @@ const MentorProfile: React.FC = () => {
...
         )}
-      <InformationModal
-        isOpen={isOpen}
-        headline="Withdraw your current application"
-        body={`Are you sure you want to withdraw your current application and apply for ${
-          mentor?.application.firstName ?? ''
-        } ${mentor?.application.lastName ?? ''}?`}
-        onConfirm={revokeApplication}
-        onClose={closeModal}
-      />
```
Comment (anjula-sack) [inline src/pages/MentorProfile/MentorProfile.component.tsx]: Why this is removed? @AnsarMahir

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/pages/MentorProfile/MentorProfile.component.tsx (hunk 7 above)
Comment (AnsarMahir) [reply src/pages/MentorProfile/MentorProfile.component.tsx]: @anjula-sack had to go back to an old commit due to removing the mentors profession. Copied that code and this was from a latest one so it also gone. Let me fix it this by today

**Comment by AnsarMahir** [comment]
@anjula-sack Hopefully done.

**Review by anjula-sack — APPROVED** [review]
Good job! @AnsarMahir

**Review by anjula-sack — CHANGES_REQUESTED** [review]
There's a lint failure

**Comment by AnsarMahir** [comment]
> There's a lint failure

@anjula-sack fixed

**Review by anjula-sack — APPROVED** [review]
Good job! @AnsarMahir

**Pull request merged**